lxd/daemon.go

```diff
@@ -1478,7 +1478,7 @@ func (d *Daemon) init() error {
 		if err == nil {
 			logger.Info("Initialized global database")
 			break
-		} else if errors.Is(err, db.ErrSomeNodesAreBehind) {
+		} else if api.StatusErrorCheck(err, http.StatusUpgradeRequired) {
```

Comment:
To clarify; we still fail on `http.StatusPreconditionFailed` (another cluster member's schema/api version was greater). So we should still expect the daemon to refuse to start until the entire cluster has been downgraded? Wouldn't that prevent the other cluster members from updating their `nodes` entries due to dqlite loss of quorum?

With members c1, c2 and c3, downgrade c1. An admin will need to power ahead with the downgrade of c2 with c1 refusing to start. Once c1 and c2 have been started, they won't be able to establish quorum (since `err` here will still be `http.StatusPreconditionFailed`), so there's no way for the third cluster member to update its schema entry.

Reply:
Not quite.

Today LXD still restarts when it is behind other cluster members, it just does so in a way that means the query it ran against the database to update its `nodes` entry with its updated version isn't persisted. This means the cluster never converges.

However the database itself is up and running and quorum has been reached, otherwise we wouldn't be able to run any queries against it.